### PR TITLE
Bug fix: Remove chargePeriod when StopDate equals single existing chargePeriod

### PR DESCRIPTION
**Pull request by HenrikSommer**
## Description

It must be possible to stop a charge with only one existing period.
To do this, we have agreed to remove the only existing period.

## References

* #1217

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs [hunk 1]
```diff
@@ -149,10 +149,10 @@ public void CancelStop(ChargePeriod chargePeriod)
 
         private void StopExistingPeriod(Instant stopDate)
         {
-            var previousPeriod = _periods
-                .SingleOrDefault(p =>
+            var previousPeriod = _periods.OrderByDescending(p => p.Description)
```
Comment (jonasdmoeller) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs:152]: Why order by description?

**Review by jonasdmoeller — CHANGES_REQUESTED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs (hunk 1 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs:152]: That's an error! Should have been StartDateTime

**Review by jonasdmoeller — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs [hunk 2]
```diff
@@ -149,10 +149,10 @@ public void CancelStop(ChargePeriod chargePeriod)
 
         private void StopExistingPeriod(Instant stopDate)
         {
-            var previousPeriod = _periods
-                .SingleOrDefault(p =>
+            var previousPeriod = _periods.OrderByDescending(p => p.StartDateTime)
+                .FirstOrDefault(p =>
                     p.EndDateTime >= stopDate &&
-                    p.StartDateTime < stopDate);
+                    p.StartDateTime <= stopDate);
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs:155]: I see a case here, that we do not handle correctly;

Two periods exist:
* Period 1: Ends at day 5
* Period 2: Starts at day 5, Ends at day 9

Stop date received = day 5

as I see it, period 2 will be selected by FirstOrDefault, after which lines 165-167 will create an incorrect newPreviousPeriod row where start and end dates are equal?

**Review by prtandrup — CHANGES_REQUESTED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs (hunk 2 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Charges/Charge.cs:155]: I agree, so I will make that change.
We still need `RemoveAllSubsequentPeriods()` for the case, when a stop is added prior to any later period(s).

**Review by prtandrup — APPROVED** [review]

**Pull request merged**